go-controller/pkg/util/net_linux.go

```diff
@@ -318,3 +319,28 @@ func DeleteConntrack(ip string, port int32, protocol kapi.Protocol) error {
 	}
 	return nil
 }
+
+// GetNetworkInterfaceIPs returns the IP addresses for the network interface 'iface'.
+func GetNetworkInterfaceIPs(iface string) ([]*net.IPNet, error) {
+	link, err := netLinkOps.LinkByName(iface)
+	if err != nil {
+		return nil, fmt.Errorf("failed to lookup link %s: %v", iface, err)
+	}
+
+	addrs, err := netLinkOps.AddrList(link, netlink.FAMILY_ALL)
+	if err != nil {
+		return nil, fmt.Errorf("failed to list addresses for %q: %v", iface, err)
+	}
+
+	var ips []*net.IPNet
+	for _, addr := range addrs {
+		if addr.Scope != unix.RT_SCOPE_UNIVERSE {
```

Comment:
this does not rule out ULA, right?

Reply:
> this does not rule out ULA, right?

It's not about the address range, it's about the flags that were used when adding it... I rewrote it to check `IsLinkLocalUnicast` instead.